src/google/protobuf/compiler/csharp/csharp_wrapper_field.cc

```diff
@@ -81,6 +81,25 @@ void WrapperFieldGenerator::GenerateMembers(io::Printer* printer) {
     "  set {\n"
     "    $name$_ = value;\n"
     "  }\n"
+    "}\n\n");
+
+  printer->Print(
```

Comment:
No need to add has method for message field in proto3. Users can compare to null.  I also think we should not add clear methods for proto3 in this PR because it is to support proto2 field presence. One of the proto3's goal is to reduce the generated code size. As users can use proto3 well with out clear method we can keep the current generated code. If you really want to add those methods, they should be another PR and need discussion for the cost and gain for decision.

Reply:
Should I also remove the default value fields from proto3 as well?

Reply:
It will be nice to keep proto3 not changed except we have to. What is the current behavior way for users to clear the field? Set to default value from the descriptor I guess?